homeassistant/components/bang_olufsen/media_player.py

```diff
@@ -620,6 +650,10 @@ async def async_select_source(self, source: str) -> None:
             # Video
             await self._client.post_remote_trigger(id=key)
 
+    async def async_select_sound_mode(self, sound_mode: str) -> None:
+        """Select a sound mode."""
+        await self._client.activate_listening_mode(id=self._sound_modes[sound_mode])
```

Comment:
I think we should catch `KeyError` and wrap in `ServiceValidationError` since the base entity does not protect against trying to set a sounds mode which does not exist: https://github.com/home-assistant/core/blob/589910b49bbd249ea1064c950aebea5be4f417c7/homeassistant/components/media_player/__init__.py#L392-L397

Reply:
I've added a check for if the sound_mode is in the integration's sound_modes.